src/modules/bootloader/main.py

```diff
@@ -637,25 +637,47 @@ def run_grub_install(fw_type, partitions, efi_directory):
                                    "--bootloader-id=" + efi_bootloader_id,
                                    "--force"])
     else:
-        assert efi_directory is None
-        if libcalamares.globalstorage.value("bootLoader") is None:
-            return
+        try:
+            install_hybrid_grub = libcalamares.job.configuration["installHybridGRUB"]
+        except KeyError:
+            install_hybrid_grub = False
 
-        boot_loader = libcalamares.globalstorage.value("bootLoader")
-        if boot_loader["installPath"] is None:
-            return
+        if efi_directory is None:
+            if libcalamares.globalstorage.value("bootLoader") is None:
+                return
+
+            boot_loader = libcalamares.globalstorage.value("bootLoader")
+            boot_loader_install_path = boot_loader["installPath"]
+            if boot_loader_install_path is None:
+                return
+        else:
+            if not install_hybrid_grub:
+                return
+
+            # We're actually on an EFI system even though fw_type is "bios".
+            # We need to find the proper drive to install to based on which
+            # drive the ESP is located on.
+            installation_root_path = libcalamares.globalstorage.value("rootMountPoint")
+            install_efi_directory = installation_root_path + efi_directory
+            find_esp_disk_command = "echo \"/dev/$(lsblk -o PKNAME $(df --output=source" \
+                + install_efi_directory \
+                + " | tail -n1) | tail -n1)\""
+            boot_loader_install_path = subprocess.run(["sh", "-c", find_esp_disk_command],
+                capture_output=True).stdout.decode().strip()
```

Comment:
Please don't use subprocess.run, instead use Calamares's python functions for running commands.

Reply:
Got it, latest commits do it the correct way.